tests/core/pyspec/eth2spec/test/capella/block_processing/test_process_deposit.py

```diff
@@ -56,3 +56,41 @@ def test_success_top_up_to_withdrawn_validator(spec, state):
         assert has_execution_withdrawal and is_withdrawable and has_non_zero_balance
     else:
         assert spec.is_fully_withdrawable_validator(validator, balance, current_epoch)
+
+
+@with_capella_and_later
+@spec_state_test
+def test_validator_invalid_high_funding_after_withdrawal(spec, state):
+    """
+     Simple extension of the previous test_success_top_up_to_withdrawn_validator in order to add a negative scenario
+     where a user deposits 2x MAX_EFFECTIVE_BALANCE.
+     Have added support for the increased Electra deposit amount
+     """
+    validator_index = 5  # Example validator index
+
+    # Fully withdraw the validator
+    set_validator_fully_withdrawable(spec, state, validator_index)
+    assert state.balances[validator_index] > 0
+
+    next_epoch_via_block(spec, state)
+    assert state.balances[validator_index] == 0
+    assert state.validators[validator_index].effective_balance > 0
+
+    next_epoch_via_block(spec, state)
+    assert state.validators[validator_index].effective_balance == 0
+
+    # Perform a check to see if the spec is Electra version
+    if is_post_electra(spec):
+        max_effective_balance = spec.MAX_EFFECTIVE_BALANCE_ELECTRA
+    else:
+        max_effective_balance = spec.MAX_EFFECTIVE_BALANCE
+
+    # Make a top-up deposit, attempting to deposit double the max effective balance
+    amount = max_effective_balance * 2
+    deposit = prepare_state_and_deposit(spec, state, validator_index, amount, signed=True)
+
+    # Process the invalid deposit, setting valid=False as we expect it to be rejected
+    run_deposit_processing(spec, state, validator_index, deposit, valid=False)
```

Comment:
So like I said before, this is a valid situation.

The invalid usage of `run_deposit_processing` is throwing an exception for the wrong reason.

It should be:

```suggestion
    # Process deposit with twice the max effective balance
    yield from run_deposit_processing(spec, state, deposit, validator_index)
```